ca/ocsp.go

```diff
@@ -243,7 +243,12 @@ func (olq *ocspLogQueue) loop() {
 					done = true
 					break inner
 				}
-				fmt.Fprintf(&builder, "%x:%d,", ol.serial, ol.status)
+				reasonStr := "_"
+				// if not good use statuscode
+				if ol.reason != 0 {
+					reasonStr = fmt.Sprintf("%d", ol.reason)
```

Comment:
Unfortunately, we have two overlapping number ranges:

status:
 - ocsp.Good ([this happens to == 0, but that's not in any IETF spec](https://pkg.go.dev/golang.org/x/crypto/ocsp#pkg-constants))
 - ocsp.Revoked
 - ocsp.Unknown

reason - https://datatracker.ietf.org/doc/html/rfc5280#page-70:

```
        unspecified             (0),
        keyCompromise           (1),
        cACompromise            (2),
        affiliationChanged      (3),
        superseded              (4),
        cessationOfOperation    (5),
        certificateHold         (6),
             -- value 7 is not used
        removeFromCRL           (8),
        privilegeWithdrawn      (9),
        aACompromise           (10) 
```

You can see that `0` is a valid `reason`. In that case we would want to log `:0`. But sometimes there is no reason because the certificate is not revoked, and we want to make that clear in the logs. So for unrevoked certificates we want to say `_`.

To implement this logic, I think the `ocspLog` struct needs to have _both_ status and reason fields.

Reply:
I am not sure if I got it 100%. Could you please have a look on my latest changes and check if it makes sense? :)